internal/cache/controller_obj_cache.go

```diff
@@ -2503,13 +2503,15 @@ func validateAndConfigureSeGroup(client *clients.AviClient) bool {
 		return false
 	}
 
-	infraSettings, err := lib.GetCRDClientset().AkoV1alpha1().AviInfraSettings().List(context.TODO(), metav1.ListOptions{})
-	if err != nil {
-		utils.AviLog.Warnf("Unable to list AviInfraSettings %s", err.Error())
-	}
 	seGroupSet := make(map[string]bool)
-	for _, setting := range infraSettings.Items {
-		seGroupSet[setting.Spec.SeGroup.Name] = true
+	if lib.GetAviInfraSettingEnabled() {
+		infraSettingList, err := lib.GetCRDClientset().AkoV1alpha1().AviInfraSettings().List(context.TODO(), metav1.ListOptions{})
+		if err != nil {
+			utils.AviLog.Warnf("Unable to list AviInfraSettings %s", err.Error())
```

Comment:
I think we should skip in case of an error.

Reply:
the items slice is empty in case of errors, so it's pretty much thee same thing as skipping the operation.

Reply:
In that case it is fine. Though there can be multiple types of errors which we probably can't test and if in any case the infraSettingList is null we might face a problem. So I thought  I though we can do an error check explicitly.